src/Plutarch/Extra/Tagged.hs

```diff
@@ -48,7 +48,56 @@ import Plutarch.TryFrom (PTryFrom (PTryFromExcess, ptryFrom'))
 import Plutarch.Unsafe (punsafeCoerce)
 import qualified PlutusTx
 
-{- | Plutarch-level 'Tagged'.
+{- | Plutarch-level 'Tagged'. @PTagged@ allows one to "tag" a given type in the typelevel.
+ Typically, @tag@ will hold `Symbol` or custom kind.  Since @PTagged@s only act as a wrapper, it does not hold any significance in the resulting script, and there are no affect on performance/script size.
+
+=== Tagged Types:
+@
+-- One can make new kind with
```

Comment:
"We can create a new kind by defining an empty type. This is then \'promoted\' by GHC to a kind."